### PR TITLE
kie-issues#1230: Rename kie-tools images from quay.io/kie-tools to docker.io/apache/incubator-kie-*

### LINKED ISSUE #1230: Rename `kie-tools` images from `quay.io/kie-tools` to `docker.io/apache/incubator-kie-*`

### LINKED ISSUE #2338: Remove dev-v1 reference in kogito-swf-devmode.feature test regarding is not valid after Quarkus 3.8
dev-v1 reference is not valid anymore after upgrading to Quarkus 3.8. This reference is used in the test:

https://github.com/apache/incubator-kie-tools/blob/699877560bc022c2d20caf1b3356b72c3f3b56ce/packages/kogito-swf-devmode/test-resources/tests/features/kogito-swf-devmode.feature#L117

It needs to be replaced by 'dev-ui' regarding the 'dev-v1'  endpoint currently  is not there in Dev mode.

### PR DESCRIPTION
**Pull request by thiagoelg**
Closes https://github.com/apache/incubator-kie-issues/issues/1230
Closes https://github.com/apache/incubator-kie-tools/issues/2338

Also fixes:
	- sonataflow-devmode-image tests
	- sonataflow-devmode-image and sonataflow-builder-image reports

### CODE REVIEW

**Comment by tiagobento** [comment]
cc @ricardozanini

**Comment by ricardozanini** [comment]
Regarding the operator, I think we should follow this PR:

- https://github.com/apache/incubator-kie-kogito-serverless-operator/pull/466

File: packages/sonataflow-devmode-image/test-resources/tests/features/sonataflow-devmode.feature [hunk 1]
```diff
@@ -112,19 +112,19 @@ Feature: Serverless Workflow devmode images
...
     Then check that page is served
-      | property             | value                                                              |
-      | port                 | 8080                                                               |
-      | path                 | /q/dev-v1/org.kie.kogito-addons-quarkus-data-index-inmemory/dataindex |
-      | request_method       | GET                                                                |
-      | wait                 | 480                                                                |
-      | expected_status_code | 200                                                                |
+      | property             | value                                                                 |
+      | port                 | 8080                                                                  |
+      | path                 | /q/dev-ui/org.kie.kogito-addons-quarkus-data-index-inmemory/dataindex |
```
Comment (nmirasch) [inline packages/sonataflow-devmode-image/test-resources/tests/features/sonataflow-devmode.feature:117]:
```suggestion
      | path                 | /q/dev-ui/org.kie.kogito-addons-quarkus-data-index-inmemory/data-index-graphql-ui |
```

Comment (nmirasch) [reply packages/sonataflow-devmode-image/test-resources/tests/features/sonataflow-devmode.feature:117]: @thiagoelg This is the dev-ui Data Index Graphql UI endpoint to check after Quarkus 3.8 update

Comment (tiagobento) [reply packages/sonataflow-devmode-image/test-resources/tests/features/sonataflow-devmode.feature:117]: @nmirasch Is this related to this PR? https://github.com/apache/incubator-kie-kogito-apps/pull/2054

Comment (nmirasch) [reply packages/sonataflow-devmode-image/test-resources/tests/features/sonataflow-devmode.feature:117]: @tiagobento yes! it adds the endpoint to current Dev-ui when data-index addon is added

Comment (tiagobento) [reply packages/sonataflow-devmode-image/test-resources/tests/features/sonataflow-devmode.feature:117]: Ok, so changed made to `kogito-apps` won't be reflected on KIE Tools until we upgrade it to the a newer timestamped SNAPSHOT.

**Review by tiagobento — APPROVED** [review]

**Review by porcelli — APPROVED** [review]

**Comment by tiagobento** [comment]
@thiagoelg I'm merging this, but let's address the fact that the base images for the CI `kie-tools-ci-build` are not going to docker yet on a separate PR.

**Review by LightGuard — APPROVED** [review]

**Pull request merged**